internal/seed/config/environment_test.go

```diff
@@ -62,8 +62,9 @@ func TestKeyMatchOverwritesValue(t *testing.T) {
 		{"generic sub", sub + "." + subKey, sub + "." + subKey, envValue, serviceName, envValue},
 		{"service root", rootKey, serviceName + "." + rootKey, envValue, serviceName, envValue},
 		{"service sub", sub + "." + subKey, serviceName + "." + sub + "." + subKey, envValue, serviceName, envValue},
+		{"service hyphen", rootKey, rootKey, envValue, "service-Name", envValue},
+		{"service underscore", rootKey, rootKey, envValue, "service_Name", envValue},
```

Comment:
These tests don't exercise the variants they describe.  Note the other preexisting service tests ("service root" and "service sub") add the serviceName in the tests.envKey value to the rootKey and sub.subKey respectively.

There are two types of overrides -- service-specific (where the serviceName is part of the environment key) and global (where it is not).  

Your two tests succeed in spite of your serviceName variations -- they match based on rootKey = rootKey (the global version).  They need to be updated for the service-specific variation (by adding the service name variation to the test's envKey).  For example:

`{"service hyphen", rootKey, "service-Name" + "." + rootKey, envValue, "service-Name", envValue},`

You'll then need to refactor your code so the resulting test does not fail.

Also, to be consistent with the existing test implementation each variation of serviceName should have two separate test cases -- one for a root key match and one for a sub key match.

Reply:
I agree, added the above test case and others too. Refactored the code as well.

From the issue description, I had gathered that we are ditching the hyphens and moving to only use the underscores, but with the context from discussion above, the description can be easily re-read to extend the naming convention to include underscores also. So this refactor includes it along with the test cases. 

Please review when you get a chance, thanks!